From a6ed21988f1477dcbc41f78a3f53c77a7cddd547 Mon Sep 17 00:00:00 2001
From: Preeti U Murthy <preetium@andrew.cmu.edu>
Date: Wed, 26 Oct 2016 03:01:02 +0530
Subject: [PATCH] Determine the offset of read and #bytes read

The offset and #bytes read are parameters that should be
passed to venus. This patch tests if the parameters in the
patch are the right ones.

---
 file.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/file.c b/file.c
index 1da3805..b4536e9 100644
--- a/file.c
+++ b/file.c
@@ -31,10 +31,16 @@ coda_file_read_iter(struct kiocb *iocb, struct iov_iter *to)
 {
 	struct file *coda_file = iocb->ki_filp;
 	struct coda_file_info *cfi = CODA_FTOC(coda_file);
+    ssize_t size;
+    
+    /* Test for 15-821 project */
+    printk("coda_read\n");
 
 	BUG_ON(!cfi || cfi->cfi_magic != CODA_MAGIC);
 
-	return vfs_iter_read(cfi->cfi_container, to, &iocb->ki_pos);
+    size = vfs_iter_read(cfi->cfi_container, to, &iocb->ki_pos);
+    printk("The read is at offset %lld, #bytes read = %ld", iocb->ki_pos, size);
+    return size;
 }
 
 static ssize_t
-- 
2.5.0

